frontend/features/page/sections/ContactsSection/ContactsSection.tsx

```diff
@@ -0,0 +1,734 @@
+                }}
+                pos="relative"
+                w="60px"
+                h="60px"
+              >
+                <GithubIcon color="white" />
+              </Splash>
+            </Box>
+          </Link>
+        </Flex>
+      </Flex>
+    </Box>
+  );
+}
+
+enum FormStatus {
+  Idle = "idle",
+  Submitting = "submitting",
+  Submitted = "submitted",
+}
+
+const blankForm = {
+  status: FormStatus.Idle,
+  values: { name: "", email: "", message: "" },
+  validationErrors: { name: null, email: null, message: null },
+  submitError: null,
+};
+
+interface UpdateFieldAction {
+  type: FormActionType.UpdateField;
+  name: string;
+  value: string;
+}
+
+interface SuccessAction {
+  type: FormActionType.Success;
+}
+
+interface FailAction {
+  type: FormActionType.Failed;
+}
+
+interface SubmitAction {
+  type: FormActionType.Submit;
+}
+
+enum FormActionType {
+  UpdateField = "update-field",
+  Success = "success",
+  Failed = "failed",
+  Submit = "submit",
+  ValidationFailed = "validation - failed",
+}
+
+interface ValidationFailed {
+  type: FormActionType.ValidationFailed;
+  validationErrors: FormState["validationErrors"];
+}
+
+type FormAction =
+  | UpdateFieldAction
+  | SuccessAction
+  | FailAction
+  | SubmitAction
+  | ValidationFailed;
+
+interface FormState {
+  status: FormStatus;
+  values: {
+    name: string;
+    email: string;
+    message: string;
+  };
+  validationErrors: {
+    name: Nullable<string>;
+    email: Nullable<string>;
+    message: Nullable<string>;
+  };
+  submitError: Nullable<string>;
+}
+
+function reducer(state: FormState, action: FormAction) {
+  switch (action.type) {
+    case FormActionType.UpdateField: {
+      return {
+        ...state,
+        values: {
+          ...state.values,
+          [action.name]: action.value,
+        },
+        validationErrors: {
+          ...state.validationErrors,
+          [action.name]: null,
+        },
+        submitError: null,
+      };
+    }
+
+    case FormActionType.Submit: {
+      //Check errors
+
+      return {
+        ...state,
+        status: FormStatus.Submitting,
+      };
+    }
+
+    case FormActionType.Success:
+      return {
+        // Display thanks page
+        ...state,
+        status: FormStatus.Submitted,
+      };
+
+    case FormActionType.Failed:
+      return {
+        // Resta in idle, setta errore e resetta i campi?
+        ...state,
+
+        submitError: "Error while sending your message",
+        status: FormStatus.Idle,
+      };
+
+    case FormActionType.ValidationFailed:
+      return {
+        ...state,
+        validationErrors: action.validationErrors,
+      };
+
+    default: {
+      assertNever(action);
+    }
+  }
+}
+
+function validateErrors(values: FormState["values"], state: FormState) {
+  const regex = /^[^@\s]+@[^@\s\.]+\.[^@\.\s]+$/;
+
+  const validationErrors: FormState["validationErrors"] = {
+    name: null,
+    email: null,
+    message: null,
+  };
```

Comment:
Having the validation object initialized to null fields only complicate things, since you then need to check if you have validation errors this way:

```
const hasErrors = Object.values(validationErrors).some(
  (prop) => prop != null
);
```

instead of a more simple `hasErrors = Object.values(validationErrors).length > 0`